watertap/unit_models/selective_oil_permeation.py

```diff
@@ -0,0 +1,669 @@
+
+        Returns: None
+        """
+        init_log = idaeslog.getInitLogger(self.name, outlvl, tag="unit")
+        solve_log = idaeslog.getSolveLogger(self.name, outlvl, tag="unit")
+
+        opt = get_solver(solver, optarg)
+
+        # ---------------------------------------------------------------------
+        # Initialize holdup block
+        flags = self.feed_side.initialize(
+            outlvl=outlvl,
+            optarg=optarg,
+            solver=solver,
+            state_args=state_args,
+        )
+        init_log.info_high("Initialization Step 1 Complete.")
+        # ---------------------------------------------------------------------
+        # Initialize permeate
+        # Set state_args from inlet state
+        if state_args is None:
+            state_args = {}
+            state_dict = self.feed_side.properties_in[
+                self.flowsheet().config.time.first()
+            ].define_port_members()
+
+            for k in state_dict.keys():
+                if state_dict[k].is_indexed():
+                    state_args[k] = {}
+                    for m in state_dict[k].keys():
+                        state_args[k][m] = state_dict[k][m].value
+                else:
+                    state_args[k] = state_dict[k].value
+
+        self.properties_permeate.initialize(
+            outlvl=outlvl,
+            optarg=optarg,
+            solver=solver,
+            state_args=state_args,
+        )
+        init_log.info_high("Initialization Step 2 Complete.")
+
+        # ---------------------------------------------------------------------
+        # Solve unit
+        with idaeslog.solver_log(solve_log, idaeslog.DEBUG) as slc:
+            res = opt.solve(self, tee=slc.tee)
+        init_log.info_high("Initialization Step 3 {}.".format(idaeslog.condition(res)))
+
+        # ---------------------------------------------------------------------
+        # Release Inlet state
+        self.feed_side.release_state(flags, outlvl + 1)
+        init_log.info("Initialization Complete: {}".format(idaeslog.condition(res)))
+
+        if not check_optimal_termination(res):
+            raise InitializationError(f"Unit model {self.name} failed to initialize")
+
+    def _get_performance_contents(self, time_point=0):
+        var_dict = {}
+        expr_dict = {}
+        var_dict["Membrane area"] = self.area
+        var_dict["Avg. transmembr. pressure"] = self.pressure_transmemb[time_point]
+        if hasattr(self, "deltaP"):
+            var_dict["Pressure drop"] = self.deltaP[time_point]
+        var_dict["Oil volumetric flux"] = self.flux_vol_oil[time_point]
+        var_dict["Pure oil volumetric flux"] = self.flux_vol_oil_pure[time_point]
+        var_dict["Effective area ratio"] = self.effective_area_ratio[time_point]
+        var_dict["Oil recovery"] = self.recovery_frac_oil[time_point]
+        var_dict["Shell side liquid velocity"] = self.liquid_velocity[time_point]
```

Comment:
Unless you change this to be velocity at inlet _and_ outlet, I'd specify this as "Shell side liquid velocity in"